docs/conf.py

```diff
@@ -64,13 +64,26 @@
 
 # -- Options for HTML output -------------------------------------------------
 
+if tags.has("external"):
+    publish_url = "docs.katanagraph.com"
+else:
+    publish_url = "docs.k9h.dev"
```

Comment:
Should this be in some function to make sure it's not executed before `tags` is populated? I really don't know, top level code is just a bit dangerous because execution order is unpredictable.

Reply:
@arthurp I think this should be ok.

Is it true that top level code execution is unpredictable in a Python script? I thought it was executed sequentially. I've not run into this before. Is there some circumstance where this happens?

Reply:
Within a module, you are correct. The issue is that module import order is hard to predict. It is technically deterministic, but it's affected by what other modules import your code, so it's really hard to predict. So the issue is whether `tags` could be changed by other code after this code executes invalidating the value of `publish_url`.

One fix would be to use a function `get_publish_url` instead. That way it would always be up to date if tags changes.

Reply:
So I just realized what file this is. Silly me. :-/ This code is fine. Sorry for the confusion.